extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQuery.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.query.scan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.druid.query.BaseQuery;
+import io.druid.query.DataSource;
+import io.druid.query.Query;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.query.filter.DimFilter;
+import io.druid.query.spec.QuerySegmentSpec;
+
+import java.util.List;
+import java.util.Map;
+
+@JsonTypeName("scan")
+public class ScanQuery extends BaseQuery<ScanResultValue>
+{
+  public static final String SCAN = "scan";
+
+  private final String resultFormat;
+  private final int batchSize;
+  private final int limit;
+  private final DimFilter dimFilter;
+  private final List<String> columns;
+
+  @JsonCreator
+  public ScanQuery(
+      @JsonProperty("dataSource") DataSource dataSource,
+      @JsonProperty("intervals") QuerySegmentSpec querySegmentSpec,
+      @JsonProperty("resultFormat") String resultFormat,
+      @JsonProperty("batchSize") int batchSize,
+      @JsonProperty("limit") int limit,
+      @JsonProperty("filter") DimFilter dimFilter,
+      @JsonProperty("columns") List<String> columns,
+      @JsonProperty("context") Map<String, Object> context
```

Comment:
we will need documentation similar to the select query on how to use the scan query